rqueue-core/src/main/java/com/github/sonus21/rqueue/listener/RqueueExecutor.java

```diff
@@ -269,14 +266,30 @@ private void execute() {
       updateCounter(true);
       failureCount += 1;
       error = e;
-    } catch (Exception e) {
+      status = ExecutionStatus.FAILED;
+    } catch (Throwable e) {
       updateCounter(true);
       failureCount += 1;
       error = e;
+      status = ExecutionStatus.FAILED;
       log(Level.ERROR, "Message execution failed, RqueueMessage: {}", e, job.getRqueueMessage());
     }
   }
 
+  private boolean shouldRetry(long maxProcessingTime, int retryCount) {
+    if (retryCount > 0 &&
+        ExecutionStatus.FAILED.equals(status) &&
+        System.currentTimeMillis() < maxProcessingTime) {
+      Set<Class<? extends Throwable>> exceptions = queueDetail.getDoNotRetry();
+      boolean doNoRetry = Objects.nonNull(exceptions) &&
+          !exceptions.isEmpty() &&
+          Objects.nonNull(error) &&
+          exceptions.contains(error.getClass());
```

Comment:
Are you sure about this? I think error.getClass() will always be a MessagingException - its cause will be the real exception (e.g. DoNotRetry2Exception)

Reply:
However, fixing this then causes the test to BootDoNotRetryTest to fail.

Reply:
ok, it's a bit more complicated:

If an exception is thrown in the handler, it will be wrapped by a MessagingException, and we have to check if `error.getCause().getClass()` is in the set of exceptions.

If an exception is thrown in middleware, then it will not (necessarily) be a MessagingException.

I could live with throwing (or re-throwing) only in the middleware, but I'm not sure if this is a good expectation.

Reply:
I see, you're right. In such cases we can check two things if it contains the actual class or it contains the cause.  That should work, let me know if you think otherwise